bin/cron/process_foorm_data

```diff
@@ -0,0 +1,98 @@
+
+SUBMISSIONS_FROM_FILE_NAME = 'submissions.csv'
+SUBMISSIONS_TO_TABLE_NAME = 'analysis.foorm_submissions_reshaped'
+FORMS_FROM_FILE_NAME = 'forms.csv'
+FORMS_TO_TABLE_NAME = 'analysis_pii.foorm_forms_reshaped'
+
+# Main method responsible for reshaping Foorm data
+# and importing to our analytics database (Redshift).
+# For each of our Foorm models (Forms and Submissions), we:
+# 1) delete any existing files (if any) from previous iterations of this job,
+# 2) reshape each record from our database into a CSV format,
+# 3) write that CSV to an S3 bucket,
+# 4) copy the contents of that CSV into a Redshift table,
+# 5) delete the CSV from S3.
+def main
+  redshift_client = RedshiftClient.instance
+
+  # Processes and uploads Foorm submissions to Redshift for analytics use via S3.
+  AWS::S3.delete_from_bucket(S3_BUCKET_NAME, SUBMISSIONS_FROM_FILE_NAME)
+  reshaped_submissions = Pd::Foorm::SubmissionAnalyticsParser.reshape_all_submissions_into_csv
+  AWS::S3.upload_to_bucket(S3_BUCKET_NAME, SUBMISSIONS_FROM_FILE_NAME, reshaped_submissions, no_random: true)
+  write_submissions_to_redshift(redshift_client, REDSHIFT_S3_ACCESS_ROLE_ARN)
+
+  # Processes and uploads Foorm forms to Redshift for analytics use via S3.
+  AWS::S3.delete_from_bucket(S3_BUCKET_NAME, FORMS_FROM_FILE_NAME)
+  reshaped_forms = Pd::Foorm::FormAnalyticsParser.reshape_all_forms_into_csv
+  AWS::S3.upload_to_bucket(S3_BUCKET_NAME, FORMS_FROM_FILE_NAME, reshaped_forms, no_random: true)
+  write_forms_to_redshift(redshift_client, REDSHIFT_S3_ACCESS_ROLE_ARN)
+end
+
+def write_forms_to_redshift(client, arn)
+  query = <<~SQL
+    DROP TABLE IF EXISTS #{FORMS_TO_TABLE_NAME};
+    CREATE TABLE #{FORMS_TO_TABLE_NAME}
+    (
+      form_id                 int,
+      form_name               varchar,
+      form_version            int,
+      question_type           varchar,
+      question_name           varchar,
+      question_text           varchar(max),
+      matrix_item_name        varchar,
+      matrix_item_text        varchar(max),
+      is_facilitator_specific int,
+      response_options        varchar(max),
+      num_response_options    int
+    );
+
+    COPY #{FORMS_TO_TABLE_NAME}
+    FROM 's3://#{S3_BUCKET_NAME}/#{FORMS_FROM_FILE_NAME}'
+    IAM_ROLE '#{arn}'
+    CSV
+    IGNOREHEADER 1;
+  SQL
+
+  client.exec(query)
+end
+
+def write_submissions_to_redshift(client, arn)
+  query = <<~SQL
+    DROP TABLE IF EXISTS #{SUBMISSIONS_TO_TABLE_NAME};
+    CREATE TABLE IF NOT EXISTS #{SUBMISSIONS_TO_TABLE_NAME}
+    (
+      submission_id    int,
+      question_name    varchar,
+      matrix_item_name varchar,
+      response_value   varchar,
+      response_text    varchar(max)
+    );
+
+    COPY #{SUBMISSIONS_TO_TABLE_NAME}
+    FROM 's3://#{S3_BUCKET_NAME}/#{SUBMISSIONS_FROM_FILE_NAME}'
+    IAM_ROLE  '#{arn}'
+    CSV
+    IGNOREHEADER 1;
+  SQL
+
+  client.exec(query)
+end
+
+begin
+  main
+ensure
+  # If processing fails at any point, delete CSVs from S3.
```

Comment:
Just checking -- if anything fails mid-way through, do we need to clean up redshift at all?

Reply:
Good thought, I hadn't really considered -- I think what we'd want to do on failure is keep the existing version of the data in Redshift (assuming the RED team has built some analytics views on top of these tables), rather than wipe it entirely.